src/InvestmentManager.sol

```diff
@@ -512,75 +541,72 @@ contract InvestmentManager is Auth {
     }
 
     // --- Helpers ---
-    function calculateDepositPrice(address user, address liquidityPool) public view returns (uint128 depositPrice) {
+    function calculateDepositPrice(address user, address liquidityPool) public view returns (uint256 depositPrice) {
         LPValues storage lpValues = orderbook[user][liquidityPool];
         if (lpValues.maxMint == 0) {
             return 0;
         }
 
-        (uint8 poolDecimals, uint8 currencyDecimals, uint8 trancheTokenDecimals) = _getPoolDecimals(liquidityPool);
-        uint128 maxDepositInPoolDecimals = _toPoolDecimals(lpValues.maxDeposit, currencyDecimals, liquidityPool);
-        uint128 maxMintInPoolDecimals = _toPoolDecimals(lpValues.maxMint, trancheTokenDecimals, liquidityPool);
-
-        depositPrice = _toUint128(
-            maxDepositInPoolDecimals.mulDiv(10 ** poolDecimals, maxMintInPoolDecimals, MathLib.Rounding.Down)
-        );
+        depositPrice = _calculatePrice(lpValues.maxDeposit, lpValues.maxMint, liquidityPool);
     }
 
-    function calculateRedeemPrice(address user, address liquidityPool) public view returns (uint128 redeemPrice) {
+    function calculateRedeemPrice(address user, address liquidityPool) public view returns (uint256 redeemPrice) {
         LPValues storage lpValues = orderbook[user][liquidityPool];
         if (lpValues.maxRedeem == 0) {
             return 0;
         }
 
-        (uint8 poolDecimals, uint8 currencyDecimals, uint8 trancheTokenDecimals) = _getPoolDecimals(liquidityPool);
-        uint128 maxWithdrawInPoolDecimals = _toPoolDecimals(lpValues.maxWithdraw, currencyDecimals, liquidityPool);
-        uint128 maxRedeemInPoolDecimals = _toPoolDecimals(lpValues.maxRedeem, trancheTokenDecimals, liquidityPool);
+        redeemPrice = _calculatePrice(lpValues.maxWithdraw, lpValues.maxRedeem, liquidityPool);
+    }
 
-        redeemPrice = _toUint128(
-            maxWithdrawInPoolDecimals.mulDiv(10 ** poolDecimals, maxRedeemInPoolDecimals, MathLib.Rounding.Down)
+    function _calculatePrice(uint128 currencyAmount, uint128 trancheTokenAmount, address liquidityPool)
+        public
+        view
+        returns (uint256 depositPrice)
+    {
+        (uint8 currencyDecimals, uint8 trancheTokenDecimals) = _getPoolDecimals(liquidityPool);
+        uint256 currencyAmountInPriceDecimals = _toPriceDecimals(currencyAmount, currencyDecimals, liquidityPool);
+        uint256 trancheTokenAmountInPriceDecimals =
+            _toPriceDecimals(trancheTokenAmount, trancheTokenDecimals, liquidityPool);
+
+        depositPrice = currencyAmountInPriceDecimals.mulDiv(
+            10 ** PRICE_DECIMALS, trancheTokenAmountInPriceDecimals, MathLib.Rounding.Down
```

Comment:
I think this still ends up being the same calculation in the end, right? Just presented in a more digestible way.

Reply:
Yes indeed, just refactored a bit.